Automattic/jetpack
# WPcom_Admin_Menu: Add special cases that don't have menus extended

#### Changes proposed in this Pull Request:
* When visiting `/wp-admin?service-worker`, do not add any actions to `admin_menu`, as all the page does is return a static js file. All work is wasted. 
  * Requests to this are handled by gutenberg
https://github.com/WordPress/gutenberg/blob/b54c2d88fa883e47822ce8088de5592ad4bacbf0/lib/pwa.php#L21-L34
* Same with `/wp-admin/index-yourstuff.php`  and `/wp-admin/index-hotstuff.php`: They do not render menus and end up going through JITM code for ultimately no effect.

For more background info, see p55Cj4-1xc-p2

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:
* Visit `/wp-admin?service-worker`. Before and after the patch, the data returned should be exactly the same.
  * BEFORE: On WPCOM, `/wp-admin?service-worker` **will** cause `get_top_messages()` in `jitm-engine.php` to run, however, the data will not be used, as there is nowhere to display the messages.
  * AFTER: `/wp-admin?service-worker` will **not** cause `get_top_messages()` in `jitm-engine.php` to run
* Same with `/wp-admin/index-yourstuff.php`  and `/wp-admin/index-hotstuff.php`
 

#### Logging screenshots before/after
 **Loading /wp-admin/ on my simple site before changes** (JITM rendered 5 times)
![2022-02-11_15-53-before](https://user-images.githubusercontent.com/937354/153675781-960f1530-f75a-4e5e-a9e6-7f447758dc57.png)

**Loading /wp-admin/ on my simple site after changes** (JITM rendered once)
![2022-02-11_15-53-after](https://user-images.githubusercontent.com/937354/153675806-a3e9de4a-0816-443e-8615-1b91d6d937e5.png)
